qsoverlay/circuit_builder.py

```diff
@@ -381,22 +361,85 @@ def update(self, **kwargs):
         for rule in self.update_rules:
             update_function_dic[rule](self, **kwargs)
 
-    def finalize(self, topo_order=False, t_add=0):
+    def finalize(self, toposort=False, tmin=None, tmax=None,
```

Comment:
I would not rename `topo_order` to `toposort` without extra need: larger chance to get reverse incompatibility in the case of someone called the function as `finalize(topo_order=True)`.